plugins/woocommerce/client/admin/client/settings-payments/components/payment-gateway-button/payment-gateway-button.tsx

```diff
@@ -15,74 +14,33 @@ export const PaymentGatewayButton = ( {
 	enabled,
 	needs_setup,
 	settings_url,
+	togglePlugin,
 	text_settings = __( 'Manage', 'woocommerce' ),
 	text_enable = __( 'Enable', 'woocommerce' ),
 	text_needs_setup = __( 'Continue setup', 'woocommerce' ),
-	setIsEnabled,
-}: Pick< PaymentGateway, 'id' | 'enabled' | 'needs_setup' | 'settings_url' > & {
+}: {
+	id: string;
+	enabled: boolean;
+	needs_setup?: boolean;
+	settings_url: string;
+	togglePlugin: ( id: string, settings_url: string ) => void;
 	text_settings?: string;
 	text_enable?: string;
 	text_needs_setup?: string;
-	setIsEnabled: ( isEnabled: boolean ) => void;
 } ) => {
-	const [ needsSetup, setNeedsSetup ] = useState( needs_setup );
 	const [ isLoading, setIsLoading ] = useState( false );
 
-	const toggleEnabled = async () => {
-		setIsLoading( true );
-
-		if ( ! window.woocommerce_admin.nonces?.gateway_toggle ) {
-			// eslint-disable-next-line no-console
-			console.warn( 'Unexpected error: Nonce not found' );
-			// Redirect to payment setting page if nonce is not found. Users should still be able to toggle the payment method from that page.
-			window.location.href = settings_url;
-			return;
-		}
-
-		try {
-			const response = await fetch( window.woocommerce_admin.ajax_url, {
-				method: 'POST',
-				headers: {
-					'Content-Type': 'application/x-www-form-urlencoded',
-				},
-				body: new URLSearchParams( {
-					action: 'woocommerce_toggle_gateway_enabled',
-					security: window.woocommerce_admin.nonces?.gateway_toggle,
-					gateway_id: id,
-				} ),
-			} );
-
-			const result = await response.json();
-
-			if ( result.success ) {
-				if ( result.data === true ) {
-					setIsEnabled( true );
-				} else if ( result.data === false ) {
-					setIsEnabled( false );
-				} else if ( result.data === 'needs_setup' ) {
-					setNeedsSetup( true );
-					window.location.href = settings_url;
-				}
-			} else {
-				window.location.href = settings_url;
-			}
-		} catch ( error ) {
-			// eslint-disable-next-line no-console
-			console.error( 'Error toggling gateway:', error );
-		} finally {
-			setIsLoading( false );
-		}
-	};
-
 	const onClick = ( e: React.MouseEvent ) => {
 		if ( ! enabled ) {
 			e.preventDefault();
-			toggleEnabled();
+			setIsLoading( true );
+			togglePlugin( id, settings_url );
```

Comment:
I think this is meant to be async/await?

Reply:
yes, it's inside the function